src/redux/handlers/onMove.js

```diff
@@ -35,25 +37,27 @@ export const onMove = (state, action) => {
     roundCounter = roundCounter + 1;
 
     players = players.map(switchFirstCardFromBuffer);
-    entities = executeMoves(
-      { ...entities, ...playersToObject(players) },
-      state.boardSize
+
+    // TODO: investigate mutations, throws error after second move
+    Immutable.setIn(
+      state,
+      ['entities'],
+      executeMoves({ ...entities, ...playersToObject(players) }, state.boardSize)
```

Comment:
przekazane entities nie posiada zmian dokonanych przez funkcje switchCardToBuffer

Reply:
Prawda, ale to i tak w tej formie nie ma sensu